dsl/Prop.cc

```diff
@@ -325,11 +361,34 @@ void Prop::patchDSL(core::MutableContext ctx, ast::ClassDef *klass) {
         }
         ENFORCE(!nodesAndPropInfo->nodes.empty(), "nodesAndPropInfo with value must have nodes be non empty");
         replaceNodes[stat.get()] = std::move(nodesAndPropInfo->nodes);
-        props.emplace_back(nodesAndPropInfo->propInfo);
+        props.emplace_back(std::move(nodesAndPropInfo->propInfo));
     }
     auto oldRHS = std::move(klass->rhs);
     klass->rhs.clear();
     klass->rhs.reserve(oldRHS.size());
+    if (synthesizeInitialize) {
+        // we define ours synthesized initialize first so that if the user wrote one themselves, it overrides ours.
+        auto loc = klass->loc;
```

Comment:
I can imagine this `loc` being super confusing in error messages. Have you tried running sorbet locally on your test cases? Do the type errors look like they could be improved? Like maybe we could use the loc of the respective prop for the loc of each arg.

Reply:
I like that idea, but what should we do for the loc of the initialize method and its sig? To me the class def seems like the only thing that makes sense.